app/schema/answer.py

```diff
@@ -70,6 +79,11 @@ def validate(self, state):
                 state.errors = []
                 state.errors.append(self.questionnaire.get_error_message('MANDATORY', self.id))
                 state.is_valid = False
+            elif self.mandatory and self.type == 'Radio' and state.value == 'other' and state.input == 'other' and state.other is None:
```

Comment:
This validation is only applicable to Radio at this point since there is a separate validation routine for a checkbox in the checkbox sub-class.